docs/fundamentals/orleans.md

```diff
@@ -0,0 +1,179 @@
+The name provided to the Orleans resource is for diagnostic purposes. For most applications, a value of `"default"` will suffice.
+
+```csharp
+var orleans = builder.AddOrleans("default");
+```
+
+The Orleans resource builder, `orleans` in the above example, offers methods to configure your Orleans resource. The simplest Orleans services only need a cluster membership resource:
+
+```csharp
+// Add a resource which Orleans supports (Azure Table Storage) to use for cluster membership:
+var storage = builder.AddAzureStorage("storage").RunAsEmulator();
+var clusteringTable = storage.AddTables("clustering");
+
+// Tell Orleans to use that resource for cluster membership:
+orleans.WithClustering(clusteringTable);
+```
+
+This tells Orleans that any service referencing it will also need to reference the `clusteringTable` resource.
+
+To participate in an Orleans cluster, reference the Orleans resource from your service project, either using `WithReference(orleans)` to participate as an Orleans server, or `WithReference(orleans.AsClient())` to participate as a client.
+When you reference the Orleans resource from your service, those resources will also be referenced:
+
+```csharp
+// Add the Orleans resource to the "silo" project:
+// As described above, this implicitly add references to the required resources.
+// In this case, that is the 'clusteringTable' resource declared earlier.
+builder.AddProject<Projects.OrleansServer>("silo")
+       .WithReference(orleans);
+```
+
+Putting that all together, here is an example of an Aspire AppHost project which includes an Orleans resource with clustering and storage, and an Orleans server and client project resource.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Add the resources which we will use for Orleans clustering and grain state storage.
+var storage = builder.AddAzureStorage("storage").RunAsEmulator();
+var clusteringTable = storage.AddTables("clustering");
+var grainStorage = storage.AddBlobs("grainstate");
+
+// Add the Orleans resource to the Aspire DistributedApplication builder, then configure it with Azure Table Storage for clustering
+// and Azure Blob Storage for grain storage.
+var orleans = builder.AddOrleans("my-app")
+                     .WithClustering(clusteringTable)
+                     .WithGrainStorage("Default", grainStorage);
+
+// Reference the Orleans resource from the 'silo' project so that it can join the Orleans cluster as a service.
+builder.AddProject<Projects.OrleansServer>("silo")
+       .WithReference(orleans)
+       .WithReplicas(3);
+
+// Reference the Orleans resource as a client from the 'frontend' project so that it can connect to the Orleans cluster.
+builder.AddProject<Projects.OrleansClient>("frontend")
+       .WithReference(orleans.AsClient())
+       .WithExternalHttpEndpoints()
+       .WithReplicas(3);
+
+// Build and run the application.
+using var app = builder.Build();
+await app.RunAsync();
```

Comment:
```suggestion
var builder = DistributedApplication.CreateBuilder(args);

// Add the resources which we will use for Orleans clustering and 
// grain state storage.
var storage = builder.AddAzureStorage("storage").RunAsEmulator();
var clusteringTable = storage.AddTables("clustering");
var grainStorage = storage.AddBlobs("grainstate");

// Add the Orleans resource to the Aspire DistributedApplication 
// builder, then configure it with Azure Table Storage for clustering
// and Azure Blob Storage for grain storage.
var orleans = builder.AddOrleans("my-app")
                     .WithClustering(clusteringTable)
                     .WithGrainStorage("Default", grainStorage);

// Reference the Orleans resource from the 'silo' project so that 
// it can join the Orleans cluster as a service.
builder.AddProject<Projects.OrleansServer>("silo")
       .WithReference(orleans)
       .WithReplicas(3);

// Reference the Orleans resource as a client from the 'frontend' 
// project so that it can connect to the Orleans cluster.
builder.AddProject<Projects.OrleansClient>("frontend")
       .WithReference(orleans.AsClient())
       .WithExternalHttpEndpoints()
       .WithReplicas(3);

// Build and run the application.
using var app = builder.Build();
await app.RunAsync();
```